### ISSUE TITLE
User authentication feature

### ISSUE THREAD

**Issue by lizhineng** [awesm/server#1]
Let's do this first, user authentication is one of the fundamental part of our app. I don't know if there is a package which contains the basic logic of user authentication that can let us to reuse it. I found a package called [Spring Security](https://spring.io/projects/spring-security) and I don't know whether we can develop our feature upon it.

On this Pull Request, maybe we could tackle the following problems:

1. Users can register through API endpoint `/auth/register`, with given only name, email, and password, then give back token if the all validation are passes.
2. Users can login through API endpoint `/auth/login`, with credential(email and password), then issue a token if the credential is valid.
3. Users can request a password recovery through `/auth/forgot-password` with given email address, and send the password recovery email to the email address.
4. With the valid password recovery token, users could reset their password through `/auth/reset-password`.

Note that all API should be prefixed with `/api`, any discussion about it are welcomed, and don't forget to build with test cases, Happy Coding :-)

**Comment by lizhishang** [comment]
yeah, i have finished it and done partial simple testing for the api. Do you have some suggestions to recode it better or do next step?

**Comment by lizhineng** [comment]
I'll look for it tonight :-)

**Comment by lizhineng** [comment]
Oh, you should create a pull request next time, not to push the code to master branch directly. @lizhishang
